### PR TITLE
Migrate devicelab tests and test runners to null safety.

### PR DESCRIPTION
**Pull request by darrenaustin**
Part of #84014.

This migrates the devicelab tests and test runners to null safety. With this everything in the device lab should be running with null safety turned on.

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Comment by darrenaustin** [comment]
Just pulled in the latest from the other PRs that landed recently.

This is still currently blocked by a bug in vm_service: https://github.com/dart-lang/sdk/issues/46559, which will have to be resolved and rolled into flutter master before this can land.

**Pull request merged**
